articles/machine-learning/service/how-to-create-your-first-pipeline.md

```diff
@@ -55,7 +55,7 @@ Create the resources required to run a pipeline:
 ### Set up a datastore
 A datastore stores the data for the pipeline to access. Each workspace has a default datastore. You can register additional datastores. 
 
-When you create your workspace, [Azure Files](https://docs.microsoft.com/azure/storage/files/storage-files-introduction) and [Azure Blob storage](https://docs.microsoft.com/azure/storage/blobs/storage-blobs-introduction) are attached to the workspace by default. Azure Files is the default datastore for a workspace, but you can also use Blob storage as a datastore. To learn more, see [Deciding when to use Azure Files, Azure Blobs, or Azure Disks](https://docs.microsoft.com/azure/storage/common/storage-decide-blobs-files-disks). 
+When you create your workspace, [Azure Files](https://docs.microsoft.com/azure/storage/files/storage-files-introduction) and [Azure Blob storage](https://docs.microsoft.com/azure/storage/blobs/storage-blobs-introduction) are attached to the workspace by default. Azure Blob Storage is the default datastore for a workspace, but you can also use Blob storage as a datastore. To learn more, see [Deciding when to use Azure Files, Azure Blobs, or Azure Disks](https://docs.microsoft.com/azure/storage/common/storage-decide-blobs-files-disks). 
```

Comment:
but you can also use azure files storage as a datastore.

Reply:
Thank you for making this change but it is still a bit confusing. It should say: 
 
When you create your workspace, a file share and a blob container are created under the azure storage account provisioned as part of workspace creation (or under the storage container provided by you). Two datastores are also attached to the workspace which stores the connection information for the blob container and file share. The datastore pointing to the blob container is also set as the default datastore and you can change the default to any other datastores under your workspace later.

Reply:
Thanks for the feedback, I have updated the text and the code block just below it as well:

When you create your workspace, [Azure File share](https://docs.microsoft.com/azure/storage/files/storage-files-introduction) and [Azure Blob storage](https://docs.microsoft.com/azure/storage/blobs/storage-blobs-introduction) are created under the azure storage account provisioned as part of workspace creation (or under the storage container provided by you). 
These two datastores are also attached to the workspace which stores the connection information for the blob container and file share. 
The datastore pointing to the blob container is also set as the default datastore and you can change the default to any other datastores under your workspace later.
```python
# Default datastore (Azure blob storage)
def_data_store = ws.get_default_datastore() 

# The above call is equivalent to this 
def_data_store = Datastore(ws, "workspaceblobstore")

# Get Azure file storage associated with the workspace
def_blob_store = Datastore(ws, "workspacefilestore")
```

Reply:
>. These two datastores are also attached to the workspace which stores the connection information for the blob container and file share.
This is not quite right. The file share and blob container are not datastores. Datastores are just pointers to storage services (e.g. blob container, file share, SQL database). Two things happen during workspace creation:
1. A storage account is provisioned
2. Workspace talks to the datastore service and asks it to create the blob container and file share under that storage account, then create two datastores pointing to the blob container and file share that was just created.
Sorry for being very nit picking, given we are trying to explain to the user what is happening, I would like to make sure that what we tell is user is precise and accurate.

Reply:
That is fine and I completely agree, but I am a bit confused now :) as that sentence was added  by you in earlier reply!

> Thank you for making this change but it is still a bit confusing. It should say:
> 
> When you create your workspace, a file share and a blob container are created under the azure storage account provisioned as part of workspace creation (or under the storage container provided by you). Two datastores are also attached to the workspace which stores the connection information for the blob container and file share. .....

Omitting it would give:

When you create your workspace, [Azure File share](https://docs.microsoft.com/azure/storage/files/storage-files-introduction) and [Azure Blob storage](https://docs.microsoft.com/azure/storage/blobs/storage-blobs-introduction) are created under the azure storage account provisioned as part of workspace creation (or under the storage container provided by you).  
The datastore pointing to the blob container is also set as the default datastore and you can change the default to any other datastores under your workspace later.